src/averageAges.js

```diff
@@ -55,7 +77,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childMotherPairs = people.map((child) => {
```

Comment:
Why not just filter mothers or sons here?

Reply:
This array will have the same length as the original but will have `false` entries. And you call `filter` again to remove them.